databus/src/main/java/com/bazaarvoice/emodb/databus/core/DefaultFanout.java

```diff
@@ -139,8 +145,10 @@ private boolean copyEvents() {
             }
 
             // Copy to subscriptions in the current data center.
-            for (Subscription subscription : _subscriptionEvaluator.matches(subscriptions, matchEventData)) {
-                eventsByChannel.put(subscription.getName(), eventData);
+            for (OwnedSubscription subscription : _subscriptionEvaluator.matches(subscriptions, matchEventData)) {
+                if (_databusAuthorizer.owner(subscription.getOwnerId()).canReceiveEventsFromTable(matchEventData.getTable().getName())) {
```

Comment:
I believe this condition should be in `SubscriptionEvaluator` class. That way no one will be able to match events without looking at the API key permissions.